pages/DSNP/UserData.md

```diff
@@ -0,0 +1,84 @@
+# User Data
+
+DSNP stores user-related data of well defined formats associated with a user's DSNP Identifier.
+Data may be public or stored in encrypted form.
+The [Avro](https://avro.apache.org) schema specification is used to define the binary representation of relevant data types.
+
+## User Data Types
+
+DSNP implementations MUST support the following user data types:
+
+| System Name | Version | Encryption Algorithm | Compression Codec | Avro Object Type |
+| --- | --- | --- | --- | --- |
+| `publicFollows` | 1.2 | NONE | [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateFollows` | 1.2 | `curve25519xsalsa20poly1305` |  [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateConnections` | 1.2 | `curve25519xsalsa20poly1305` | [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateConnectionPRIds` | 1.2 | NONE | NONE | [PRId](Graph.md#pseudonymous-relationship-identifiers) |
+
+Data for each data type is initially formatted as a stream of Avro objects that should conform to the schema specified.
+Avro file- and block-level information (including in-stream schema) is omitted.
+The Avro stream is then compressed (or not) and encrypted (or not) as specified.
+
+`curve25519xsalsa20poly1305` (that is, X25519 key exchange, XSalsa20 encryption, and Poly1305 message authentication) is the default authenticated encryption algorithm used in the [NaCl](https://nacl.cr.yp.to) ("Salt") library, and its successor [libsodium](https://libsodium.org). In the specification of cryptographic operations below, relevant methods from these libraries are noted. While these specific libraries are not required for DSNP compatibility, they are highly recommended.
+
+## Data Chunks
+
+Because blockchain systems often have specific limits to the amount of data that can be included in a given transaction, operations on user data deal with the data in discrete chunks.
+As implementation strategies may vary, implementations MUST define their own maximum chunk size in bytes to be used in the operations described below.
+
+## Entity Tags
+
+To allow for application-layer caching and prevent race conditions between different applications attempting to update the same data, an implementation MUST provide an `etag` (entity tag) value with each chunk fetched, and update this for each replaced chunk.
+An implementation MUST fail the Replace User Data Operation if the `etag` values supplied do not match the current `etag` values for all chunks of the specified data type.
+
+## Replace User Data Operation
+
+The Replace User Data Operation takes the following parameters:
+
+* The DSNP User ID of the user who controls the data
```

Comment:
Trying out a different wording. The current one might be best, but it reads confusing to me.
```suggestion
* The DSNP User ID for the data
```

Reply:
Going to change this to `A DSNP User Id` because I think it should be obvious from the context (and the following line about delegation).